tt_metal/hostdevcommon/api/hostdevcommon/flags.hpp

```diff
@@ -0,0 +1,35 @@
+// SPDX-FileCopyrightText: © 2025 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <type_traits>
+
+template <typename E>
+struct Flags {
```

Comment:
```suggestion
class Flags {
```
I would prefer change struct to class and scopes everything cleanly under public and private.